src/javis_viewer.jl

```diff
@@ -0,0 +1,230 @@
+    # via signal_connect. Otherwise, we will be unable to update the
+    # widget that is automatically created by the slider object.
+    #
+    # It should be stated that a `slider` object is essentially a
+    # GtkScale widget coupled with a Reactive object.
+    #
+    =#
+
+    # Create a textbox
+    tbox = GtkReactive.textbox(Int; signal = signal(slide))
+
+    # Button for going forward through animation
+    forward = GtkButton("==>")
+
+    # Button for going backward through animation
+    backward = GtkButton("<==")
+
+    #=
+
+    TODO: Enable widgets of window to dynamically resize based on user changing the size of a window.
+
+    I think I can use the `configure-event` signal in GTK3 documentation (link: https://developer.gnome.org/gtk3/stable/GtkWidget.html#GtkWidget-configure-event). From there, I can then make a `signal_connect` set-up where I update `set_gtk_property!()` of the windows accordingly using `:width_request` and `height_request`.
+
+    =#
+
+    #####################################################################
+    # VIEWER CANVAS AND GRID CONFIGURATION
+    #####################################################################
+
+    # Gtk Canvas object upon which to draw image; sized via frame size
+    canvas = Gtk.Canvas(frame_dims[1], frame_dims[2])
+
+    # Grid to allocate widgets
+    grid = Gtk.Grid()
+
+    # Allocate the widgets in a 3x3 grid
+    grid[1:3, 1] = canvas
+    grid[1:3, 2] = slide
+    grid[1, 3] = backward
+    grid[2, 3] = tbox
+    grid[3, 3] = forward
+
+    # Center all widgets vertically in grid
+    set_gtk_property!(grid, :valign, 3)
+
+    # Center all widgets horizontally in grid
+    set_gtk_property!(grid, :halign, 3)
+
+    # Adds grid to previously defined window
+    push!(win, grid)
+
+    #####################################################################
+    # DISPLAY FIRST FRAME
+    #####################################################################
+
+    _draw_image(video, action_list, 1, canvas, frame_dims)
+
+    #####################################################################
+    # SIGNAL CONNECTION FUNCTIONS
+    #####################################################################
+
+    # When the slider is changed, update currently viewed frame
+    signal_connect(_slide, "value-changed") do widget
+        # Collects GtkScale as an adjustable bounded value object
+        bound_slide = Gtk.GAccessor.adjustment(_slide)
+
+        # Get frame number from bounded value object as Int
+        slide_val = Gtk.get_gtk_property(bound_slide, "value", Int)
+
+        _draw_image(video, action_list, slide_val, canvas, frame_dims)
+    end
+
+    # When the `Enter` key is pressed, update the frame
+    signal_connect(win, "key-press-event") do widget, event
+        if event.keyval == 65293
+            # Get current frame from textbox as an Int value
+            curr_frame = parse(Int, get_gtk_property(tbox, :text, String))
```

Comment:
I think `clamp` will solve this issue such that you don't need the `if elseif else` 
Btw I would call it `nframes` if it's the number of total frames

Reply:
Or of course `total_nframes` :smile:

Reply:
I love `clamp`! Thanks for telling me about it! Also, changed it to total frames. :)